avro-fastserde/build.gradle

```diff
@@ -145,3 +151,111 @@ cleanupAndRebuildTestsForAvro18.dependsOn generateAvroClasses18
+
+task sourceJar(type: Jar) {
+  from sourceSets.main.allJava
+  classifier "sources"
+}
+
+task javadocJar(type: Jar) {
+  from javadoc
+  classifier = 'javadoc'
+}
+
+task testJar(type: Jar) {
+  from sourceSets.test.allJava
+  classifier = 'tests'
+}
+
+publishing {
+  publications {
+    "$project.name"(MavenPublication) {
+      groupId project.group
+      artifactId project.name
+      version project.version
+
+      from components.java
+      artifact sourceJar
+      artifact javadocJar
+      artifact testJar
+
+      //we strive to meet https://central.sonatype.org/pages/requirements.html
+      pom {
+        name = 'Avro Util'
+        description = 'utilities for writing code that works across major avro versions'
+        url = 'https://github.com/linkedin/avro-util'
+
+        licenses {
+          license {
+            name = 'BSD 2-Clause'
+            url = 'https://raw.githubusercontent.com/linkedin/avro-util/master/LICENSE'
+          }
+        }
+        developers {
+          developer {
+            id = 'radai-rosenblatt'
+            name = 'Radai Rosenblatt'
+            email = 'radai.rosenblatt@gmail.com'
+            organization = 'LinkedIn'
+            organizationUrl = 'linkedin.com'
+          }
+          developer {
+            id = 'abhishekmendhekar'
+            name = 'Abhishek Mendhekar'
+            organization = 'LinkedIn'
+            organizationUrl = 'linkedin.com'
+          }
+          developer {
+            id = 'jimhe'
+            name = 'Jim He'
+            email = 'jimjhe@gmail.com'
+            organization = 'LinkedIn'
+            organizationUrl = 'linkedin.com'
+          }
+          developer {
+            id = 'ghosthack'
+            name = 'Adrian Fernandez'
+            email = 'adrian@ghosthack.com'
+            organization = 'LinkedIn'
+            organizationUrl = 'linkedin.com'
+          }
+        }
+        scm {
+          connection = 'scm:git:git://github.com:linkedin/avro-util.git'
+          developerConnection = 'scm:git:ssh://github.com:linkedin/avro-util.git'
+          url = 'https://github.com/linkedin/avro-util'
+        }
+      }
+
+      //remove the dependency on helper, and replace with one on helper-all
+      pom.withXml {
+        groovy.util.Node dependenciesNode = (groovy.util.Node) (asNode().dependencies[0])
+        Collection<groovy.util.Node> dependencyNodes = dependenciesNode.children();
+        for (groovy.util.Node o : dependencyNodes) {
+          if ("$project.group" == o.groupId[0].text()) {
+            dependencyNodes.remove(o)
+            break;
+          }
+        }
+        def dependencyNode = dependenciesNode.appendNode('dependency')
+        dependencyNode.appendNode('groupId', "$project.group")
+        dependencyNode.appendNode('artifactId', "helper-all")
+        dependencyNode.appendNode('version', "$project.version")
+        dependencyNode.appendNode('scope', "compile")
```

Comment:
Nitpick: `compile` is deprecated in modern versions of Gradle, in favor of `implementation` and `api`. In most cases, `implementation` is the right scope to use.

Reply:
this is a maven scope though